### PR TITLE
compilation: virt: Make libvirt support optional

### PR DESCRIPTION
**Pull request by roidelapluie**
refs #114

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by mildred** [comment]
Out of curiosity, what does `ldd mgmt` says without libvirt?

**Comment by roidelapluie** [comment]
Well  it does not show libvirt

**Comment by roidelapluie** [comment]
Wow! Huge difference!

Without libvirt:

```
	linux-vdso.so.1 (0x00007ffd49884000)
	libpthread.so.0 => /usr/host/lib/libpthread.so.0 (0x00007fe3a5c97000)
	libdl.so.2 => /usr/host/lib/libdl.so.2 (0x00007fe3a5a93000)
	libc.so.6 => /usr/host/lib/libc.so.6 (0x00007fe3a56f1000)
	/usr/x86_64-pc-linux-gnu/lib/ld-linux-x86-64.so.2 (0x00007fe3a5eb4000)
```




With libvirt:

```
	linux-vdso.so.1 (0x00007ffe5afdb000)
	libvirt-qemu.so.0 => /usr/host/lib/libvirt-qemu.so.0 (0x00007fa088fdd000)
	libvirt.so.0 => /usr/host/lib/libvirt.so.0 (0x00007fa088b3a000)
	libpthread.so.0 => /usr/host/lib/libpthread.so.0 (0x00007fa08891d000)
	libdl.so.2 => /usr/host/lib/libdl.so.2 (0x00007fa088719000)
	libc.so.6 => /usr/host/lib/libc.so.6 (0x00007fa088377000)
	libcap-ng.so.0 => /usr/x86_64-pc-linux-gnu/lib/libcap-ng.so.0 (0x00007fa088172000)
	libyajl.so.2 => /usr/x86_64-pc-linux-gnu/lib/libyajl.so.2 (0x00007fa087f69000)
	libnl-route-3.so.200 => /usr/x86_64-pc-linux-gnu/lib/libnl-route-3.so.200 (0x00007fa087cff000)
	libnl-3.so.200 => /usr/x86_64-pc-linux-gnu/lib/libnl-3.so.200 (0x00007fa087ae0000)
	libxml2.so.2 => /usr/x86_64-pc-linux-gnu/lib/libxml2.so.2 (0x00007fa08777c000)
	libcurl.so.4 => /usr/x86_64-pc-linux-gnu/lib/libcurl.so.4 (0x00007fa08750f000)
	libdbus-1.so.3 => /usr/x86_64-pc-linux-gnu/lib/libdbus-1.so.3 (0x00007fa0872c1000)
	libsystemd.so.0 => /usr/x86_64-pc-linux-gnu/lib/libsystemd.so.0 (0x00007fa087040000)
	libcap.so.2 => /usr/x86_64-pc-linux-gnu/lib/libcap.so.2 (0x00007fa086e3b000)
	librt.so.1 => /usr/x86_64-pc-linux-gnu/lib/librt.so.1 (0x00007fa086c33000)
	libm.so.6 => /usr/x86_64-pc-linux-gnu/lib/libm.so.6 (0x00007fa086936000)
	libresolv.so.2 => /usr/x86_64-pc-linux-gnu/lib/libresolv.so.2 (0x00007fa08671f000)
	liblzma.so.5 => /usr/x86_64-pc-linux-gnu/lib/liblzma.so.5 (0x00007fa0864fa000)
	libgnutls.so.30 => /usr/x86_64-pc-linux-gnu/lib/libgnutls.so.30 (0x00007fa0861de000)
	libidn.so.11 => /usr/x86_64-pc-linux-gnu/lib/libidn.so.11 (0x00007fa085fab000)
	libtasn1.so.6 => /usr/x86_64-pc-linux-gnu/lib/libtasn1.so.6 (0x00007fa085d99000)
	libnettle.so.6 => /usr/x86_64-pc-linux-gnu/lib/libnettle.so.6 (0x00007fa085b63000)
	libhogweed.so.4 => /usr/x86_64-pc-linux-gnu/lib/libhogweed.so.4 (0x00007fa08592f000)
	libgmp.so.10 => /usr/x86_64-pc-linux-gnu/lib/libgmp.so.10 (0x00007fa0856b9000)
	libgcrypt.so.20 => /usr/x86_64-pc-linux-gnu/lib/libgcrypt.so.20 (0x00007fa0853ae000)
	libgpg-error.so.0 => /usr/x86_64-pc-linux-gnu/lib/libgpg-error.so.0 (0x00007fa08519c000)
	libssh2.so.1 => /usr/x86_64-pc-linux-gnu/lib/libssh2.so.1 (0x00007fa084f6f000)
	libssl.so.1.0.0 => /usr/x86_64-pc-linux-gnu/lib/libssl.so.1.0.0 (0x00007fa084d02000)
	libcrypto.so.1.0.0 => /usr/x86_64-pc-linux-gnu/lib/libcrypto.so.1.0.0 (0x00007fa0848c9000)
	libz.so.1 => /usr/x86_64-pc-linux-gnu/lib/libz.so.1 (0x00007fa0846b4000)
	libutil.so.1 => /usr/x86_64-pc-linux-gnu/lib/libutil.so.1 (0x00007fa0844b1000)
	/usr/x86_64-pc-linux-gnu/lib/ld-linux-x86-64.so.2 (0x00007fa0891e1000)
```


operating system: exherbo linux

File: docs/documentation.md [hunk 1]
```diff
@@ -545,6 +547,18 @@ like to try. The canonical example is when running `mgmt` with `--remote` there
 might be a cached copy of the binary in the primary prefix, but in case there's
 no binary available continue working in a temporary directory to avoid failure.
 
+### Compilation option
+
+You can control some compilation variables by using environment variable.
```
Comment (purpleidea) [inline docs/documentation.md]: variables*

File: docs/documentation.md [hunk 2]
```diff
@@ -545,6 +547,18 @@ like to try. The canonical example is when running `mgmt` with `--remote` there
 might be a cached copy of the binary in the primary prefix, but in case there's
 no binary available continue working in a temporary directory to avoid failure.
 
+### Compilation option
```
Comment (purpleidea) [inline docs/documentation.md]: options*

File: resources/virt_disabled.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// +build novirt
+
+package resources
+
+// VirtRes represents the fields of the Virt resource. Since this file is
+// only invoked with the tag "novirt", we do not need any field here.
```
Comment (purpleidea) [inline resources/virt_disabled.go]: fields*

**Pull request merged**